### ISSUE TITLE
Uninitialized Memory Exposure

### ISSUE THREAD

**Issue by Ilshidur** [mhart/StringStream#7]
[According to Snyk](https://snyk.io/vuln/npm:stringstream:20180511), _stringstream_ has a vulnerability when run on Node.js 4.x and below.

The bug comes from this line : https://github.com/mhart/StringStream/blob/v0.0.5/stringstream.js#L32
More details can be found here : https://hackerone.com/reports/321670

Thank you for the package, by the way !

**Comment by knoxcard** [comment]
@Ilshidur - love how you remembered to say thanks, I remember to do the same when I submit my pull requests.  Important to first show our appreciate the efforts the maintainers put forth on these projects.

**Comment by HugoGiraudel** [comment]
Hello.

This is also reported by [nsp](https://nodesecurity.io/advisories/664) and currently shows up in `npm` and `nsp` audits.

Is there an easy way to fix this? :)

**Comment by HugoGiraudel** [comment]
From what I understand from [Snyk’s report](https://snyk.io/vuln/npm:stringstream:20180511), it should be fixable by stringifying `data` in case it’s a number?

```diff
  if (this.fromEncoding) {
    if (Buffer.isBuffer(data)) data = data.toString()
+   if (typeof data === 'number') data = String(data)
    data = new Buffer(data, this.fromEncoding)
  }
```

Or did I miss something?

**Comment by mhart** [comment]
It's about the second argument to the Buffer constructor – just need to enforce that it's undefined/null or a string (or not a number)

**Comment by mhart** [comment]
You're absolutely right – my bad. Sorry, I got confused when I was reviewing #8 – which misses the Buffer case

**Comment by mhart** [comment]
@HugoGiraudel #9 should fix it – let me know if you spot any issues with it

**Comment by HugoGiraudel** [comment]
Perfect, thank you! :)

**Comment by marian-r** [comment]
@mhart NSP doesn't recognize this as patched https://nodesecurity.io/advisories/664. Has that been patched and if yes how could we mark it as patched?

**Comment by mhart** [comment]
Yes it's been patched – that's what this issue is about. I don't know anything about nodesecurity.io – maybe @Ilshidur who reported this does?

**Comment by Ilshidur** [comment]
@mhart

I don't really know about the validation process of nsp and Snyk. Sadly, I couldn't find any indication about marking this package as patched.

However, I sometimes witnessed package authors notifying the Snyk team about their fix.
I guess the only way to find out is to contact the nsp/snyk team.

As you are the maintainer of lots of projects (some of which I love !), I can contact the respective teams about this fix if you want.

**Comment by mhart** [comment]
Looks like it was @chalker who reported it on hackerone.com – with @lirantal confirming

If one of them could close it, that would be easiest
